directives/static/js/thebe.js

```diff
@@ -144,3 +151,51 @@ var detectLanguage = (language) => {
     }
     return language;
 }
+
+
+// A function to check whether a kernel is still connected
+// A native solution would be better, but I didn't see one
+const connectionChecker = (previousInterval, previouslyConnected) => {
+    const time_to_failure_ms = 2000 
+    const check_interval_ms = 10000
+
+    if (previousInterval) clearInterval(previousInterval)
+
+    const timer = () => (
+        new Promise((res) => {
+            setTimeout(() => res(false), time_to_failure_ms)
+        })
+    )
+
+    // Promise that checks whether kernel returns info
+    const ack = () => (
+        window.thebeKernel
+        .requestKernelInfo()
+        .then(() => true)
+        .catch(() => false)
+    )
+
+    const timeOutCheck = () => Promise.race([timer(), ack()])
+
+    intervalId = window.setInterval( (previouslyConnected) => {
+        timeOutCheck().then((connected) => {
+            if (!connected && previouslyConnected) {
+            console.log(`Kernel disconnected`)
```

Comment:
This if body is not indented. The `else if` is on the same line as the last contents of the if body. I would start a new line with the closing brace of the if block and the `else if` line: 
`} else if (connected && ....) {`